llama_stack/distribution/datatypes.py

```diff
@@ -63,6 +63,15 @@ class AccessAttributes(BaseModel):
     )
 
 
+class AccessAttributesRule(BaseModel):
```

Comment:
@grs what do you think about adopting a Cedar (AWS) style simple rule engine here. 

- You have a sequence of rules one for each resource type. So all rules aren't mashed up together. 
- Each rule is something like 
```
permit|forbid ( principal … , action … , resource … )
    when { … }       # optional attribute checks (ABAC)
    unless { … };    # optional deny-unless override
```
- they are executed in order

An example of a rule could be: 

```
comment: "ashwin shouldnt commit",
rule: forbid (
  principal == "ashwin", 
  action == "commit",
  resource == "github"
) when (
  branch == "main"
)
``` 

The critical part of this language is the use of explicit "verbs" so we can read out the rule and know how the semantic works instantly.

Reply:
That certainly looks very expressive. Were you thinking to have rules in that format inline in the yaml (as multiline strings?)?

Reply:
or define policy in a separate file? Would this then obviate the need for access attributes on resources?

Reply:
> That certainly looks very expressive. Were you thinking to have rules in that format inline in the yaml (as multiline strings?)?

No, multiline strings will cause much pain. But we can make that structured. Like:
```yaml
forbid:
  - ("eq", "principal", "ashwin")
  - ("eq", "action", "commit")
when:
  - ..
 ```
something like that.

Defining the policy in a separate YAML or JSON file would also be fine (and likely better thing to do.) And yes, this will obviate the need for access attributes written _into_ resources? Maybe not completely because we still need to know who wrote a resource right? How do we check the rule "let all users in team X or project Y access resources created by users in project Z or namespace Foo"?